packages/headers-parser/src/results.ts

```diff
@@ -0,0 +1,27 @@
+import type { Header } from './types.js'
+
+// If one header fails to parse, we still try to return the other ones
+export function splitResults<Type>(results: (Error | Type)[]) {
+  const headers = results.filter((result) => !isError(result)) as Type[]
+  const errors = results.filter(isError) as Error[]
+  return { headers, errors }
+}
+
+const isError = function (result: Error | any) {
+  return result instanceof Error
+}
+
+// Concatenate an array of `{ headers, erors }`
+export const concatResults = function (resultsArrays: { headers: Header[]; errors: Error[] }[]) {
+  const headers = resultsArrays.flatMap(getHeaders) as Header[]
+  const errors = resultsArrays.flatMap(getErrors) as Error[]
+  return { headers, errors }
+}
+
+const getHeaders = function ({ headers }) {
```

Comment:
Totally boils down to personal preference, but it feels like this and `getErrors` could be inlined in `concatErrors`, since we've inlined [more complex functions](https://github.com/netlify/build/pull/4653/files#diff-e4912d784cfc1d181c03b6ea8a77486e3b4f24071fe808669ebe2a497f6e4475R5)?